src/pages/EditResourcePage.tsx

```diff
@@ -18,29 +22,71 @@ const StyledEditPublication = styled.div`
 `;
 
 interface EditResourcePageParamTypes {
-  identifier: string;
+  resourceIdentifierFromParam: string;
 }
 
 const EditResourcePage: FC = () => {
-  const { identifier } = useParams<EditResourcePageParamTypes>();
-  const [expanded, setExpanded] = useState<string | false>(false);
   const { t } = useTranslation();
+  const { resourceIdentifierFromParam } = useParams<EditResourcePageParamTypes>();
+
+  const [resourceIdentifier, setResourceIdentifier] = useState<string>(resourceIdentifierFromParam);
+  const [expanded, setExpanded] = useState<string | false>(false);
+  const [showForm, setShowForm] = useState<boolean>(!!resourceIdentifier);
+  const [resourceType, setResourceType] = useState<ResourceCreationType>(ResourceCreationType.FILE);
+
+  const onCreateFile = (resourceIdentifierCreatedOnMainFileSelection: string) => {
+    if (resourceIdentifierCreatedOnMainFileSelection) {
+      setResourceIdentifier(resourceIdentifierCreatedOnMainFileSelection);
+      setResourceType(ResourceCreationType.FILE);
+      setShowForm(true);
+    }
+  };
+
+  const onSubmitLink = (resourceIdentifier: string) => {
+    setResourceType(ResourceCreationType.LINK);
+    setResourceIdentifier(resourceIdentifier);
+    setShowForm(true);
+  };
+
+  const mainFileHandler = useUppy('', false, onCreateFile);
 
   const handleChange = (panel: string) => (_: React.ChangeEvent<any>, isExpanded: boolean) => {
     setExpanded(isExpanded ? panel : false);
   };
 
-  return !identifier ? (
+  //triggers on uppy-events
+  useEffect(() => {
+    if (mainFileHandler) {
+      mainFileHandler.on('upload', (file, response) => {
+        setResourceType(ResourceCreationType.FILE);
+      });
+      mainFileHandler.on('upload-error', () => {
+        toast.error('File upload error');
+      });
+    }
```

Comment:
disse kan være litt skumle siden de kan legge til samme handler flere ganger (hver gang useEffecten trigges). det var derfor vi dro inn den stygge `hasUploadSuccessEventListener`, for å sjekke om vi allerede har lagt til den eventen. pass på at dette ikke kjøres flere ganger om man feks går til en annen side før man går tilbake igjen (siden dette alltid vil kjøre på onMount)

Reply:
jeg tror ikke vi får samme problemet med "upload"-eventen, men legger på upload-error-eventen